CMakeLists.txt

```diff
@@ -90,18 +89,30 @@ SET(BOOST_OPTIONAL_COMPONENTS "")
 IF(BUILD_PYTHON_INTERFACE)
   SET(BOOST_OPTIONAL_COMPONENTS ${BOOST_OPTIONAL_COMPONENTS} python)
   FINDPYTHON()
-  ADD_COMPILE_DEPENDENCY("eigenpy >= 2.2.0")
-  INCLUDE_DIRECTORIES(SYSTEM ${PYTHON_INCLUDE_DIRS})
+  ADD_PROJECT_DEPENDENCY(eigenpy 2.2.0 REQUIRED PKG_CONFIG_REQUIRES "eigenpy >= 2.2.0")
 ENDIF(BUILD_PYTHON_INTERFACE)
 
 SET(BOOST_COMPONENTS ${BOOST_REQUIERED_COMPONENTS} ${BOOST_OPTIONAL_COMPONENTS} ${BOOST_BUILD_COMPONENTS})
 SEARCH_FOR_BOOST()
 
-# Path to boost headers
-INCLUDE_DIRECTORIES(SYSTEM ${Boost_INCLUDE_DIRS})
+# Main library
+FILE(GLOB_RECURSE ${PROJECT_NAME}_SOURCES
+  src/*.cpp
+  )
 
-# Include and add the main directory
-ADD_SUBDIRECTORY(src)
+IF(UNIX)
+  ADD_LIBRARY(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_SOURCES})
+  SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX)
+  TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME} PUBLIC $<INSTALL_INTERFACE:include>)
+  TARGET_LINK_LIBRARIES(${PROJECT_NAME} pinocchio::pinocchio
```

Comment:
@jcarpent : do you mean here ?
This is the case by default : https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-both-a-target-and-its-dependents

Reply:
What I meant is that some stuff should be public, some others not.